File: src/minStackIB.java
class Solution {
    //create constructor
    public Solution() {
        this.stack = new Stack<>();
        this.minStack = new Stack<>();
    }

    public void push(int x) {
    stack.push(x);
    if(minStack.isEmpty())
        minStack.push(x);
    else{
        if(minStack.top()<x)
            minStack.push(minStack.top());
        else 
            minStack.push(x);
    }
}

    public void pop() {
        if(!minStack.isEmpty() && !stack.isEmpty()){
            minStack.pop();
            stack.pop();
        }
    }

    public int top() {
        if(!stack.isEmpty()){
            return stack.top();
        }
        else return -1;
    }

    public int getMin() {
        if(!minStack.isEmpty()){
            return minStack.pop();
        } 
        else return -1;
    }
}

File: src/minStackIB.py
class MinStack:
    def __init__(self):
        self.stack = list()
        self.minstack = list()

    def push(self, val):
        self.stack.append(val)
        
        if self.minstack:
            if self.minstack[-1] < val:
                self.minstack.append(self.minstack[-1])
            else:
                self.minstack.append(val)
        else:
            self.minstack.append(val)

    def pop(self):
        if self.stack and self.minstack:
            del self.stack[-1]
            del self.minstack[-1]
        

    def top(self):
        return self.stack[-1] if self.stack else -1
        

    def getMin(self):
        if self.minstack:
            return self.minstack[-1]
        else: return -1
        